### PR TITLE
feat: disable webln create invoice button on invalid amount

### LINKED ISSUE #380: Normal Swaps: Disable "Create Invoice via WebLN" button when amounts are invalid
image

### PR DESCRIPTION
**Pull request by dni**
closes #380

### CODE REVIEW

File: src/components/WeblnButton.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    const createWeblnInvoice = async () => {
+        enableWebln(async () => {
+            const amount = Number(receiveAmount());
+            const invoice = await window.webln.makeInvoice({ amount: amount });
+            log.debug("created webln invoice", invoice);
+            setInvoice(invoice.paymentRequest);
+        });
+    };
+    return (
+        <Show when={webln()}>
```
Comment (michael1011) [inline src/components/WeblnButton.tsx:21]: Components themselves should not decide if they are rendered. The parent does that

File: src/components/WeblnButton.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            <button
+                id="webln"
+                disabled={!sendAmountValid()}
+                class="btn btn-light"
+                onClick={() => createWeblnInvoice()}>
+                {t("create_invoice_webln")}
+            </button>
+            <hr class="spacer" />
```
Comment (michael1011) [inline src/components/WeblnButton.tsx:29]: Same for the surrounding styling

File: tests/components/WeblnButton.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { render } from "@solidjs/testing-library";
+
+import WeblnButton from "../../src/components/WeblnButton";
+import { contextWrapper } from "../helper";
+
+describe("WeblnButton", () => {
```
Comment (michael1011) [inline tests/components/WeblnButton.spec.tsx:6]: Also test the disabling logic

**Review by michael1011 — CHANGES_REQUESTED** [review]

**Review by kilrau — CHANGES_REQUESTED** [review]
- didn't you want to leave it enabled for the default empty state because [one can type amounts](https://github.com/BoltzExchange/boltz-web-app/issues/380#issuecomment-1921466443) ?
- if amount is not default empty, "out of min/max" error should have prio over invoice error
Screenshot from 2024-02-12 19-25-38

**Comment by dni** [comment]
> didn't you want to leave it enabled for the default empty state because

it is enabled when amount are empty

**Comment by kilrau** [comment]
> > didn't you want to leave it enabled for the default empty state because
> 
> it is enabled when amount are empty

No it isn't.

image

**Comment by kilrau** [comment]
> * if amount is not default empty, "out of min/max" error should have prio over invoice error

still not fixed
image

**Pull request merged**